docs/src/main/asciidoc/mongodb-panache-guide.adoc

```diff
@@ -0,0 +1,525 @@
+That makes for very concise but yet readable code.
+MongoDB native queries are also supported.
+
+== Solution
+
+We recommend that you follow the instructions in the next sections and create the application step by step.
+However, you can go right to the completed example.
+
+Clone the Git repository: `git clone {quickstarts-clone-url}`, or download an {quickstarts-archive-url}[archive].
+
+The solution is located in the `using-mongodb-panache` {quickstarts-tree-url}/using-mongodb-panache[directory].
+
+== Creating the Maven project
+
+First, we need a new project. Create a new project with the following command:
+
+[source,shell,subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=using-mongodb-panache \
+    -DclassName="org.acme.rest.json.FruitResource" \
+    -Dpath="/fruits" \
+    -Dextensions="resteasy-jsonb,mongodb-panache"
+----
+
+This command generates a Maven structure importing the RESTEasy/JAX-RS, JSON-B and MongoDB with Panache extensions.
+After this, the quarkus-mongodb-panache extension has been added to your `pom.xml`.
+
+If you don't want to generate a new project you can add the dependency in your `pom.xml`:
+
+[source,xml]
+--
+<dependencies>
+    <dependency>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-mongodb-panache</artifactId>
+    </dependency>
+</dependencies>
+--
+
+== Setting up and configuring MongoDB with Panache
+
+To get started:
+
+* add your settings in `{config-file}`
+* Make your entities extend `PanacheMongoEntity`, you can use the `@MongoEntity` annotation to specify the name of the database and the name of the collection (it will default to the name of your entity).
+* place your entity logic in static methods in your entities
+
+Then add the relevant configuration properties in `{config-file}`.
+
+[source,properties]
+--
+# configure the MongoDB client for a replica set of two nodes
+quarkus.mongodb.connection-string = mongodb://mongo1:27017,mongo2:27017
+# mandatory if you don't specify the name of the database using @MongoEntity
+quarkus.mongodb.database = person
+--
+
+The `quarkus.mongodb.database` property will be used by MongoDB with Panache to determine the name of the database where your entities will be persisted.
+
+For advanced configuration of the MongoDB client, you can follow the link:mongo-guide.html#configuring-the-mongodb-database[Configuring the MongoDB database].
+
+== Defining your entity
+
+To define a Panache entity, simply extend `PanacheMongoEntity` and add your columns as public fields.
+You can add the `@MongoEntity` annotation to your entity if you need to customize the name of the collection and/or the database.
+
+[source,java]
+--
+@MongoEntity(collection="ThePerson")
+public class Person extends PanacheMongoEntity {
+    public String name;
+    // will be persisted as a 'birth' field in MongoDB
+    @BsonProperty("birth") public LocalDate birthDate;
```

Comment:
My intuition is that it would be a lot more readable on 2 separate lines (and with a new line between each property)

Reply:
Done.